romancal/tweakreg/tweakreg_step.py

```diff
@@ -134,460 +141,331 @@ def process(self, input):
                     else:
                         images.shelve(model, i, modify=False)
 
+        # set path where the source catalog will be saved to
         if len(self.catalog_path) == 0:
             self.catalog_path = os.getcwd()
-
         self.catalog_path = Path(self.catalog_path).as_posix()
         self.log.info(f"All source catalogs will be saved to: {self.catalog_path}")
 
-        if self.abs_refcat is None or len(self.abs_refcat.strip()) == 0:
-            self.abs_refcat = DEFAULT_ABS_REFCAT
-
+        # set reference catalog name
+        if not self.abs_refcat:
+            self.abs_refcat = DEFAULT_ABS_REFCAT.strip().upper()
         if self.abs_refcat != DEFAULT_ABS_REFCAT:
-            # Set expand_refcat to True to eliminate possibility of duplicate
-            # entries when aligning to absolute astrometric reference catalog
             self.expand_refcat = True
 
-        if len(images) == 0:
-            raise ValueError("Input must contain at least one image model.")
-
-        # Build the catalogs for input images
+        # build the catalogs for input images
+        imcats = []
         with images:
             for i, image_model in enumerate(images):
-                if image_model.meta.exposure.type != "WFI_IMAGE":
-                    # Check to see if attempt to run tweakreg on non-Image data
+                exposure_type = image_model.meta.exposure.type
+                if exposure_type != "WFI_IMAGE":
```

Comment:
The current implementation loops over the input and if the exposure type is a non-WFI_IMAGE we set `meta.cal_step` to `"SKIPPED"` and then returns a `ModelLibrary` object.

Reply:
I opened a PR with a test that fails with this PR. I don't know if this is a representative test (it has 2 WFI_IMAGE and 1 WFI_GRISM members) but I hope it illustrates the problem. The test fails on the call to `del image_model.meta.source_detection["tweakreg_catalog"]` because it attempts to delete `tweakreg_catalog` from the grism observation (and it didn't add `tweakreg_catalog` earlier in the step).

Reply:
Can you remind me, Mairan, if we need to support passing spectroscopic data to tweakreg at present?  e.g., because the regtests send it, or because of the issue you identified in the ELP where it's passing more than it needs to.  I agree with Brett that morally we should probably just error out immediately if we ever see spectroscopic data passed to tweakreg, but we don't need that in this PR necessarily.

Reply:
> Can you remind me, Mairan, if we need to support passing spectroscopic data to tweakreg at present?

Not to my knowledge. For example, the ELP pipeline behavior is to check for the exposure type, and if it is not a `"WFI_IMAGE"` then it sets `meta.cal_step.tweakreg` to `"SKIPPED"` and then added to a list that will be passed to `TweakRegStep` at the end. There's a [note in the ELP code](https://github.com/spacetelescope/romancal/blob/96af8bbc6a6dafd48d1beb251b5209d9da875ca1/romancal/pipeline/exposure_pipeline.py#L183-L186) that reads:
```
# Now that all the exposures are collated, run tweakreg
# Note: this does not cover the case where the asn mixes imaging and spectral
#          observations. This should not occur on-prem
```

Reply:
For now, `TweakRegStep` will always return all the datamodels passed on to it (even mixed exposure types). However, if the exposure is of type "WFI_IMAGE", the step will process it normally. Otherwise, it will set `meta.cal_step="SKIPPED"` and nothing else. But all the updated input files will still be returned to the caller.